src/Security/Voter/Committee/CommitteeCandidacyVoter.php

```diff
@@ -3,28 +3,65 @@
 namespace App\Security\Voter\Committee;
 
 use App\Entity\Adherent;
+use App\Entity\Committee;
 use App\Security\Voter\AbstractAdherentVoter;
+use App\VotingPlatform\Designation\DesignationTypeEnum;
 
 class CommitteeCandidacyVoter extends AbstractAdherentVoter
 {
     public const PERMISSION = 'ABLE_TO_CANDIDATE';
 
+    /**
+     * @param Committee $subject
+     */
     protected function doVoteOnAttribute(string $attribute, Adherent $adherent, $subject): bool
     {
-        if (
-            $adherent->isReferent()
-            || $adherent->isSupervisor()
-            || $adherent->isDeputy()
-            || $adherent->isSenator()
-        ) {
+        if (!($membership = $adherent->getMembershipFor($subject)) || !($election = $subject->getCurrentElection())) {
             return false;
         }
 
-        return true;
+        if (DesignationTypeEnum::COMMITTEE_ADHERENT === $election->getDesignationType()) {
+            if (
+                $adherent->isReferent()
+                || $adherent->isSupervisor()
+                || $adherent->isDeputy()
+                || $adherent->isSenator()
+            ) {
+                return false;
+            }
+
+            return true;
```

Comment:
```suggestion
            return
                !$adherent->isReferent()
                && !$adherent->isSupervisor()
                && !$adherent->isDeputy()
                && !$adherent->isSenator()
            ;
```

Reply:
yes it's an alternative, but I found more readable with IF and without NOT operator